### PR TITLE
Adjust the Jenkins E2E_REPORT_DIR to something less silly

### PR DESCRIPTION
**Pull request by zmerlynn**
Start of a fix for complaints on #8852.

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @ixdy

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -231,3 +233,10 @@ if [[ "${E2E_DOWN,,}" == "true" ]]; then
     sleep 30
     go run ./hack/e2e.go ${E2E_OPT} -v --down
 fi
+
+# TODO(zml): We have a bunch of legacy Jenkins configs that are
+# expecting junit.xml to be in ${WORKSPACE} root and it's Friday
+# afternoon, so just put the junit report where it's expected.
+if [ -e ${ARTIFACTS}/junit.xml ]; then
+  cp ${ARTIFACTS}/junit.xml ${WORKSPACE}
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: make this a symlink, and move it before cleanup so that it runs even if teardown fails?

**Comment by zmerlynn** [comment]
PTAL

File: hack/jenkins/e2e.sh [hunk 2]
```diff
@@ -220,6 +222,13 @@ if [[ "${E2E_TEST,,}" == "true" ]]; then
     go run ./hack/e2e.go ${E2E_OPT} -v --test --test_args="${GINKGO_TEST_ARGS}--ginkgo.noColor" || true
 fi
 
+# TODO(zml): We have a bunch of legacy Jenkins configs that are
+# expecting junit.xml to be in ${WORKSPACE} root and it's Friday
+# afternoon, so just put the junit report where it's expected.
+if [ -e ${ARTIFACTS}/junit.xml ]; then
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: actually, I realize now that this is not going to work - Ginkgo outputs junit%2d.xml files.

Comment (ixdy) [inline hack/jenkins/e2e.sh]: rather, make that `junit_%02d.xml`.

Comment (zmerlynn) [inline hack/jenkins/e2e.sh]: It's almost like there were junit*.xml patterns sitting right in front of my eyes.

**Comment by zmerlynn** [comment]
PTYAL.

**Comment by ixdy** [comment]
LGTM finally

**Pull request merged**